src/chainstate/stacks/db/blocks.rs

```diff
@@ -3080,25 +3098,15 @@ impl StacksChainState {
         }
     }
 
-    /// Get the highest processed block.  Break ties on lexigraphical ordering of the block hash
+    /// Get the highest processed block on the canonical burn chain.
+    /// Break ties on lexigraphical ordering of the block hash
     /// (i.e. arbitrarily).  The staging block will be returned, but no block data will be filled
     /// in.
-    pub fn get_stacks_chain_tip(&self) -> Result<Option<StagingBlock>, Error> {
-        let sql = "SELECT * FROM staging_blocks WHERE processed = 1 AND orphaned = 0 ORDER BY height DESC, anchored_block_hash DESC LIMIT 1";
-        let mut rows = query_rows::<StagingBlock, _>(&self.blocks_db, &sql, NO_PARAMS).map_err(Error::DBError)?;
-        let len = rows.len();
-        match len {
-            0 => {
-                Ok(None)
-            }
-            1 => {
-                Ok(Some(rows.pop().unwrap()))
-            },
-            _ => {
-                // should be impossible since this is the primary key
-                panic!("Got two or more block rows with same burn and block hashes");
-            }
-        }
+    pub fn get_stacks_chain_tip(&self, burndb: &BurnDB) -> Result<Option<StagingBlock>, Error> {
+        let (burn_bhh, block_bhh) = BurnDB::get_canonical_stacks_chain_tip_hash(burndb.conn())?;
+        let sql = "SELECT * FROM staging_blocks WHERE processed = 1 AND orphaned = 0 AND burn_header_hash = ?1 AND anchored_block_hash = ?2";
```

Comment:
Definitely a separate issue, but should we think about doing a pass and adding indexes on fields in the different tables / dbs?

Reply:
I can add an index over `processed` and `orphaned`.

Reply:
Fixed in 90c7c96